CRM/Core/SiteTokens.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+use Civi\Token\AbstractTokenSubscriber;
+use Civi\Token\TokenRow;
+
+/**
+ * Class CRM_Site_Tokens
+ *
+ * Generate "site.*" tokens.
+ */
+class CRM_Core_SiteTokens extends AbstractTokenSubscriber {
+  /**
+   * @var string
+   *   Token prefix
+   */
+  public $entity = 'site';
+
+  /**
+   * @var array
+   *   List of tokens provided by this class
+   *   Array(string $fieldName => string $label).
+   */
+  public $tokenNames;
+
+  /**
+   * Class constructor.
+   */
+  public function __construct() {
+    parent::__construct($this->entity, $this->getSiteTokens());
+  }
+
+  public function getSiteTokens(): array {
+    $ret = [];
+    $siteTokens = \Civi\Api4\SiteToken::get(TRUE)
+      ->addSelect('name', 'label')
+      ->addWhere('domain_id', '=', 'current_domain')
+      ->addWhere('is_active', '=', TRUE)
+      ->execute();
+    foreach ($siteTokens as $siteToken) {
+      $ret[$siteToken['name']] = $siteToken['label'];
+    }
+    return $ret;
+  }
+
+  /**
+   * @inheritDoc
+   * @throws \CRM_Core_Exception
+   */
+  public function evaluateToken(TokenRow $row, $entity, $field, $prefetch = NULL): void {
+    $row->format('text/html')->tokens($entity, $field, self::getSiteTokenValues()[$field]);
+    $row->format('text/plain')->tokens($entity, $field, self::getSiteTokenValues(NULL, FALSE)[$field]);
+  }
+
+  /**
+   * Get the site tokens available for the domain.
+   *
+   * This function will be made protected soon... FIXME: this line is copied from CRM_Core_DomainTokens::getDomainTokenValues, so not sure if it's true here. -TwoMice
+   *
+   * @param int|null $domainID
+   * @param bool $html
+   *
+   * @return array
+   * @throws \CRM_Core_Exception
+   * @internal
+   *
+   * @todo - make this non-static & protected. Remove last deprecated fn that calls it.
+   */
+  public static function getSiteTokenValues(?int $domainID = NULL, bool $html = TRUE): array {
```

Comment:
```suggestion
  private function getSiteTokenValues(?int $domainID = NULL, bool $html = TRUE): array {
```

this was only ever public / static on domain tokens as part of the transition  from `CRM_Core_Token` - make everything private that does not need to be public or else when we want to alter them someone will have some weird & wonderful code calling it directly & will think it's our fault if it breaks